### DISCUSSION TITLE
Does Firebase Dynamic Links works from a separated module?

### DISCUSSION THREAD

**Discussion by mihaicris-adoreme** [firebase/firebase-ios-sdk#10364] [General]
I am currently migrating from monolith app to iOS main target + several swift modules.
I have moved all the usages of Firebase to a dedicated module`FirebasePlatfomLive`(which imports the Firebase SDK) and a `FirebasePlatfom` interface module which exposes some public methods.  `FirebasePlatfomLive` implements `FirebasePlatfom`.
I am facing an issue with Firebase Dynamic Links product which doesn't recognise the URLs with associated domains defined in the main target app (containing the Info.plist).  (applinks:your_dynamic_links_domain)

This method always fails although I am sending a corect firebase dynamic link.
```swift

  if let dynamicLink = DynamicLinks.dynamicLinks().dynamicLink(fromCustomSchemeURL: url) {
    // Handle the deep link. For example, show the deep-linked content or
    // apply a promotional offer to the user's account.
    // ...
  }

```

Why is that? Is there a way to use Firebase Dynamic Links from another module, not from the main app target?

**Comment by morganchen12** [comment]
Hey @mihaicris-adoreme, how is `FirebasePlatformLive` linked? Is it static or dynamic?

**Reply by mihaicris-adoreme** [reply]
Hi @morganchen12, static.

**Reply by morganchen12** [reply]
From my knowledge there's nothing module-specific in Dynamic Links. @eldhosembabu do you know what could be going on here?

**Comment by mihaicris-adoreme** [comment]
Ok, found the problem.

I was using a private property reference in the service class like

```swift
private let dynamicLinksService = DynamicLinks.dynamicLinks()
```
 
and use it later in a method like:

```swift
func ...() {
    dynamicLinksService.handleUniversalLink(url) ...
}
```

Debugging the code reveal some description for the instance  `DynamicLinks` as `uninitialised`.
Using it as 

```swift
DynamicLinks.dynamicLinks().handleUniversalLink(url)...
``` 
fixed the issue.

Not sure why it is not working in the first place.
Might be converted to an issue and closed.
Thanks
